cleansio/censor/censor_realtime_mac.py

```diff
@@ -0,0 +1,190 @@
+
+            if processing_queue_length >= 2:
+                print('processing_queue length='+str(processing_queue_length))
+                print('index={}'.format(index))
+                with self.processing_lock:
+                    print(len(self.processing_queue))
+                    frames_to_process = self.processing_queue.pop(0)
+                with self.processing_lock:
+                    print(len(self.processing_queue))
+                    next_frames = self.processing_queue[0]
+
+                # Convert next two recordings into chunks
+                recorded_chunk, file_path = self.__convert_frames_to_chunk(frames_to_process, index)
+                next_recorded_chunk, _ = self.__convert_frames_to_chunk(next_frames, (index+1)*5)
+
+                overlapping_chunk, overlapping_path = self.__create_overlapping_chunk(recorded_chunk, next_recorded_chunk, file_path)
+
+                # Create accuracy chunk for current chunk and overlapping chunk
+                self.__create_accuracy_chunk(recorded_chunk, file_path)
+                self.__create_accuracy_chunk(overlapping_chunk, overlapping_path)
+
+                # censor current chunk and also mute any spillover explicits from previous chunk
+                clean_chunk_wrapper = self.censor_audio_chunk(file_path)
+                clean_chunk = AudioSegment.silent(duration=leftover_mute) + clean_chunk_wrapper.segment[leftover_mute:]
+
+                # remember to mute any overlapping explicit in the next chunk
+                leftover_mute = clean_chunk_wrapper.mute_next_start
+                if leftover_mute > 0: print("Remember to mute " + str(leftover_mute) + "s at the start of the next chunk")
+
+                # Convert current chunk into frames and add it to the playback queue
+                clean_frames = self.__convert_clean_chunk_to_frames(clean_chunk)
+                with self.playback_lock:
+                    self.playback_queue.append(clean_frames)
+                
+                if self.args.store_recording:
+                    self.clean_file += clean_chunk
+
+                index += 1
+
+    def __convert_frames_to_chunk(self, frames, index):
+        file_path = self.chunk_prefix + str(index) +'.wav'
+        sf.write(file_path, frames, self.samplerate)
+        self.__update_env_chunks_list(file_path)
+        recorded_chunk = read_and_convert_audio(file_path)
+        return recorded_chunk, file_path
+
+    def __convert_clean_chunk_to_frames(self, chunk):
+        clean_chunk_filepath = self.directory + 'clean_chunk.wav'
+        chunk.export(clean_chunk_filepath, format='wav')
+        clean_frames, _ = sf.read(clean_chunk_filepath, dtype='float32', fill_value=0.0, frames=int(self.samplerate*self.duration), always_2d=True)
+        return clean_frames
+
+    def __create_overlapping_chunk(self, chunk1, chunk2, file_path):
+        overlapping_chunk = chunk1[2500:] + chunk2[:2500]
```

Comment:
For whatever reason, this results in an overlapping_chunk that is chunk1[2500:] + **chunk1**[:2500]. It makes absolutely no sense, and I can't figure out why its happening

Reply:
Resolved by using indata.copy()